server/dumpbagserver/views.py

```diff
@@ -5,55 +5,73 @@
 from datetime import datetime
 
 from dumpbagserver import app, app_config
-from flask import render_template, redirect, url_for, flash, jsonify
+from flask import flash, jsonify, redirect, render_template, request, url_for
 from werkzeug.routing import UnicodeConverter, ValidationError
 
 from .bagger import Bagger
+from .forms import SearchForm
 
 
 class DBNameConverter(UnicodeConverter):
-
     def to_python(self, value):
         if not re.match("^[A-Za-z0-9_-]+$", value):
             raise ValidationError()
         return value
 
 
-app.url_map.converters['dbname'] = DBNameConverter
+app.url_map.converters["dbname"] = DBNameConverter
 
 
-RE_DUMP_DATE = re.compile(r'.*(\d{8}-\d{6}).*')
+RE_DUMP_DATE = re.compile(r".*(\d{8}-\d{6}).*")
 
 
-@app.route('/databases')
+@app.route("/databases")
 def databases():
     databases = Bagger(app_config).list_databases()
-    return render_template('databases.html', databases=databases)
+    return render_template("databases.html", databases=databases)
 
 
-@app.route('/')
+@app.route("/")
 def dumps():
     bagger = Bagger(app_config)
+    search_form = SearchForm(request.args)
     dumps = bagger.list_dumps()
-    return render_template('dumps.html', dumps=dumps,
-                           download_commands=bagger.download_commands)
-
-
-@app.route('/dump/<dbname:dbname>')
+    if request.args.get("search", type=str) and search_form.validate():
+        sname = request.args.get("search", type=str)
+        res = dict()
+        for dbs in dumps:
+            if sname in dbs:
+                res[dbs] = dumps[dbs]
+        return render_template(
+            "dumps.html",
+            dumps=res,
+            download_commands=bagger.download_commands,
+            form=search_form,
+        )
+    else:
+        return render_template(
+            "dumps.html",
+            dumps=dumps,
+            download_commands=bagger.download_commands,
+            form=search_form,
+        )
```

Comment:
the duplication of the call to `render_template` is not necessary:

```python
if request.args.get("search", type=str) and search_form.validate():
    sname = request.args.get("search", type=str)
    dumps = {dbname: value for dbname, value in dumps.copy().items() if sname in dbname}
return render_template(
    "dumps.html",
    dumps=dumps,
    download_commands=bagger.download_commands,
    form=search_form,
)
```

Reply:
@guewen Is is done, thanks.

Reply:
@guewen We added also your suggestion with the javascript. It is very much appreciated, thank you.

Reply:
hmm but what's the reason to have both the server-side search *and* the js search? My remark was that using a server-side search is overkill for the need and less ergonomic

Reply:
I point out that my js code was incomplete (https://github.com/camptocamp/odoo-dump-bag/pull/17#issuecomment-650166475 for what's missing), mainly, it should hide the main titles of databases when no database is found. But if you need help on that, let me know.

Reply:
Having both makes the service degrade gracefully (and doesn't hurt otherwise).

Reply:
I would say that every line of code hurts a bit (code is a liability), but whatever, if you want.
Still, note that as of today, if you disable javascript, you can't "unroll" the sections, so you won't go far. A fix could be to show everything by default when js is disabled, but then CTRL-F would be a good enough graceful search :stuck_out_tongue_closed_eyes: